beaconChain/lib/forkChoice/lmdGhost.ts

```diff
@@ -0,0 +1,100 @@
+import {Attestation, BeaconBlock, BeaconState, PendingAttestation, Validator, ValidatorIndex} from "../../types";
+import {getActiveValidatorIndices, getEffectiveBalance, slotToEpoch} from "../../helpers/stateTransitionHelpers";
+import {FORK_CHOICE_BALANCE_INCREMENT} from "../../constants";
+import {Slot} from "../../types";
+
+type int = number;
+
+// Probably add this as a field to BeaconState
+interface Store {
+  blocks: BeaconBlock[];
+  pendingAttestations: PendingAttestation[];
+  validatorRegistry: Validator[];
+}
+
+interface AttestationTarget {
+  validatorIndex: ValidatorIndex;
+  target: PendingAttestation;
+}
+
+/**
+ * Get the ancestor of block with slot number slot; return None if not found.
+ * @param {Store} store
+ * @param {BeaconBlock} block
+ * @param {slot} slot
+ * @returns {BeaconBlock}
+ */
+function getAncestor(store: Store, block: BeaconBlock, slot: Slot): BeaconBlock {
+  if (block.slot === slot) {
+      return block;
+  } else if (block.slot < slot) {
+    return null;
+  } else {
+    // TODO Find way to access parent block properly
+    return getAncestor(store, store.blocks.pop(), slot);
```

Comment:
probably should stub to just select the last block rather than remove it from the store.